### PR TITLE
[5.6] make model factory after method signatures more consistent

### PR DESCRIPTION
**Pull request by unstoppablecarl**
This does some "breaking" cleanup for #23551

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Factory.php [hunk 1]
```diff
@@ -116,11 +116,12 @@ public function state($class, $state, $attributes)
      *
      * @param  string  $class
      * @param  callable  $callback
+     * @param  string  $name
      * @return $this
      */
-    public function afterMaking($class, $callback)
+    public function afterMaking($class, callable $callback, $name = 'default')
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Factory.php]: Please remove this extra type from the definition. It's a breaking change to the signature.

File: src/Illuminate/Database/Eloquent/Factory.php [hunk 2]
```diff
@@ -133,23 +134,22 @@ public function afterMaking($class, $callback)
      * @param  callable  $callback
      * @return $this
      */
-    public function afterMakingState($class, $state, $callback)
+    public function afterMakingState($class, $state, callable $callback)
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Factory.php]: Same

File: src/Illuminate/Database/Eloquent/Factory.php [hunk 3]
```diff
@@ -133,23 +134,22 @@ public function afterMaking($class, $callback)
...
     /**
      * Define a callback to run after creating a model.
      *
      * @param  string  $class
      * @param  callable  $callback
+     * @param  string $name
      * @return $this
      */
-    public function afterCreating($class, $callback)
+    public function afterCreating($class, callable $callback, $name = 'default')
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Factory.php]: etc

File: src/Illuminate/Database/Eloquent/Factory.php [hunk 4]
```diff
@@ -162,11 +162,9 @@ public function afterCreating($class, $callback)
      * @param  callable  $callback
      * @return $this
      */
-    public function afterCreatingState($class, $state, $callback)
+    public function afterCreatingState($class, $state, callable $callback)
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Factory.php]: If you want to add this, please do it on master.

Comment (unstoppablecarl) [reply src/Illuminate/Database/Eloquent/Factory.php]: I did and @taylorotwell said to put it here. https://github.com/laravel/framework/pull/23670

Comment (devcircus) [reply src/Illuminate/Database/Eloquent/Factory.php]: The feature has not been tagged yet so no worries about breaking changes.

Comment (GrahamCampbell) [reply src/Illuminate/Database/Eloquent/Factory.php]: Ah, right, good.

**Pull request merged**